include/seqan3/range/views/take_until.hpp

```diff
@@ -69,281 +69,14 @@ class view_take_until : public std::ranges::view_interface<view_take_until<urng_
-            base_t{std::move(it)}
-        {}
-
-        //!\brief Constructor that delegates to the CRTP layer and initialises the callable.
-        iterator_type(base_base_t it,
-                      fun_ref_t _fun,
-                      sentinel_type /*only used by the consuming iterator*/) noexcept(noexcept(base_t{it})) :
-            base_t{std::move(it)}, fun{_fun}
-        {}
-        //!\}
-
-        /*!\name Associated types
-         * \brief All are derived from the base_base_t.
-         * \{
-         */
-
-        //!\brief The difference type.
-        using difference_type       = typename std::iterator_traits<base_base_t>::difference_type;
-        //!\brief The value type.
-        using value_type            = typename std::iterator_traits<base_base_t>::value_type;
-        //!\brief The reference type.
-        using reference             = typename std::iterator_traits<base_base_t>::reference;
-        //!\brief The pointer type.
-        using pointer               = typename std::iterator_traits<base_base_t>::pointer;
-        //!\brief The iterator category tag.
-        using iterator_category     = iterator_tag_t<base_base_t>;
-        //!\}
-
-        /*!\name Comparison operators
-         * \brief We define comparison against self and against the sentinel.
-         * \{
-         */
-        //!\brief Delegate comparison to base_base_t.
-        bool operator==(iterator_type const & rhs) const
-            noexcept(noexcept(std::declval<base_base_t &>() == std::declval<base_base_t &>()))
-        //!\cond
-            requires std::forward_iterator<base_base_t>
-        //!\endcond
-        {
-            return *this->this_to_base() == *rhs.this_to_base();
-        }
-
-        //!\brief Evaluate functor, possibly throw.
-        bool operator==(sentinel_type const & rhs) const
-            noexcept(!or_throw &&
-                     noexcept(std::declval<base_base_t &>() == std::declval<sentinel_type &>()) &&
-                     noexcept(fun(std::declval<reference>())))
-        {
-            if (*this->this_to_base() == rhs) // [[unlikely]]
-            {
-                if constexpr (or_throw)
-                    throw unexpected_end_of_input{"Reached end of input before functor evaluated to true."};
-                else
-                    return true;
-            }
-
-            return fun(**this);
-        }
+    class iterator_type;
 
-        //!\brief Switch lhs and rhs for comparison.
-        friend bool operator==(sentinel_type const & lhs, iterator_type const & rhs)
-            noexcept(noexcept(rhs == lhs))
-        {
-            return rhs == lhs;
-        }
-
-        //!\brief Switch lhs and rhs for comparison.
-        bool operator!=(sentinel_type const & rhs) const
-            noexcept(noexcept(std::declval<iterator_type &>() == rhs))
-        {
-            return !(*this == rhs);
-        }
+    template <bool is_const_range>
+    class take_until_sentinel;
```

Comment:
I added https://github.com/orgs/seqan/projects/4#card-38949132 to have consistent naming.